craft_parts/dirs.py

```diff
@@ -64,14 +64,16 @@ def __init__(
             self._partitions = None
             self.partition_dir = None
 
-        self.stage_dirs = MappingProxyType(
-            partition_utils.get_partition_dir_map(
-                base_dir=self.work_dir, partitions=partitions, suffix="stage"
-            )
-        )
-        self.prime_dirs = MappingProxyType(
+        self.overlay_dirs = self._get_partition_dirs(partitions, "overlay")
```

Comment:
Overlay dirs for partitions is used to save the overlay migration state file. These can be moved to the state dir in a separate refactoring.